pkg/steps/pod_test.go

```diff
@@ -182,3 +184,135 @@ func TestPodStepExecution(t *testing.T) {
 		})
 	}
 }
+
+func TestGetPodObject(t *testing.T) {
+	testCases := []struct {
+		name        string
+		podStep     func(*podStep)
+		expectedPod func(*v1.Pod)
+	}{
+		{
+			name: "no secret",
+			podStep: func(expectedPodStepTemplate *podStep) {
+				expectedPodStepTemplate.config.SecretName = ""
+			},
+			expectedPod: func(expectedPodTemplate *v1.Pod) {
+				expectedPodTemplate.Spec.Volumes = []v1.Volume{}
+				expectedPodTemplate.Spec.Containers[0].VolumeMounts = []v1.VolumeMount{}
+			},
+		},
+		{
+			name: "with test secret",
+			podStep: func(expectedPodStepTemplate *podStep) {
+				expectedPodStepTemplate.config.SecretName = testSecretName
+			},
+			expectedPod: func(expectedPodTemplate *v1.Pod) {},
+		},
+		{
+			name: "with test secret and path",
+			podStep: func(expectedPodStepTemplate *podStep) {
+				expectedPodStepTemplate.config.SecretName = testSecretName
+				expectedPodStepTemplate.config.SecretPath = "/usr/local/secrets"
+			},
+			expectedPod: func(expectedPodTemplate *v1.Pod) {
+				expectedPodTemplate.Spec.Containers[0].VolumeMounts = []coreapi.VolumeMount{
+					{
+						Name:      testSecretName,
+						ReadOnly:  true,
+						MountPath: "/usr/local/secrets",
+					},
+				}
+			},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			podStepTemplate := expectedPodStepTemplate()
+			podTemplate := expectedPodTemplate()
+			tc.podStep(podStepTemplate)
+			tc.expectedPod(podTemplate)
+			pod, err := podStepTemplate.getPodObject()
+			if err != nil {
+				t.Errorf("test case %s error %v", tc.name, err)
+			}
+
+			if !reflect.DeepEqual(pod, podTemplate) {
+				//if true {
+				t.Errorf("test %s failed. generated pod spec was not as expected", tc.name)
+				t.Error(pod)
+				t.Error(podTemplate)
+			}
+		})
+	}
+
+}
+
+func expectedPodTemplate() *v1.Pod {
```

Comment:
If you only check the fields you care about your test will not need this, will not be convoluted with other behavior of the system and will be more concise

Reply:
having a hard time to imagine how this could be squashed.

Reply:
None of your tests care about the labels, or the annotations, or the command .. or ... or ... 

If you follow my suggestion and don't use `DeepEqual` but instead validate the specific fields you care about (feel free to `DeepEqual` on the  volume mounts, if you want) you don't need to present any extraneous fields and you don't need this func